### ISSUE TITLE
Contacts Page

### ISSUE THREAD

**Issue by francis-specialeffect** [sandstorm-io/sandstorm#2832]
Currently the only way to view your contacts as far as I can see is to 'Share Grain' and 'Send an invite'. This is fine and works, but I feel like it could be improved. My suggestion would be to have a contacts page of sorts on the sidebar, where you can view and manage everyone with whom you've shared your identity. This would also make it possible to add contacts without sharing a grain via link, which is just slightly neater and more intuitive in my opinion. You could also from the contacts page, have the ability to remove contacts, and share grains from there, though that might be overly complicating things.

**Comment by kentonv** [comment]
Yeah, I'd love to have a contact list built into the UI somewhere, allowing you to see who is online, message people, or quickly create a new grain with someone (maybe that's how messages would work! it would be an app).

Big project, though. :)

**Comment by francis-specialeffect** [comment]
Yeah I like the sound of incorporating messages into one contacts/messaging app, a lot of work as you say, but quite a bit of the functionality is already there, in that you can add contacts, albeit indirectly, and a list of them is stored somewhere.
